src/internal/types.test.ts

```diff
@@ -156,3 +161,144 @@ describe("IfBoundedRecord", () => {
     >().toEqualTypeOf<false>();
   });
 });
+
+describe("EnumerableStringKeyOf", () => {
+  test("string keys", () => {
+    expectTypeOf<
+      EnumerableStringKeyOf<Record<string, unknown>>
+    >().toEqualTypeOf<string>();
+  });
+
+  test("number keys", () => {
+    expectTypeOf<
+      EnumerableStringKeyOf<Record<number, unknown>>
+    >().toEqualTypeOf<`${number}`>();
+  });
+
+  test("union of records", () => {
+    expectTypeOf<
+      EnumerableStringKeyOf<Record<number, unknown> | Record<string, unknown>>
```

Comment:
Instead of testing number and string, which cause typescript to squish the result type, you can use a template strings which won't be squished with the number string:

```suggestion
      EnumerableStringKeyOf<Record<number, unknown> | Record<`prefix_${string}`, unknown>>
```